### PR TITLE
Add speedrun notifications

### PR DESCRIPTION
**Pull request by Mm2PL**
Needs a tiny bit of name changes in the code

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 1]
```diff
@@ -392,4 +399,34 @@ default String clueNotifyMessage() {
...
+    }
+
+    @ConfigItem(
+        keyName = "speedrunSendImage",
+        name = "Send Image",
+        description = "Send image with the notification",
+        position = 33,
+        section = clueSection
+    )
+    default boolean speedrunSendImage() { return true; }
```
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:431]: re-order these 2 so it follows the order implemented in #17

Comment (Mm2PL) [reply src/main/java/dinkplugin/DinkPluginConfig.java]: image
fixed in 8731d59

**Review by Felanbird — APPROVED** [review]

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 2]
```diff
@@ -69,6 +70,7 @@ public class DinkPlugin extends Plugin {
     public static final Pattern COLLECTION_LOG_REGEX = Pattern.compile("New item added to your collection log: (?<itemName>(.*))");
     private static final Pattern PET_REGEX = Pattern.compile("You have a funny feeling like you.*");
 
+
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPlugin.java:73]: We have another PR touching nearby code, it'd be best to get rid of this change

```suggestion
```

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 3]
```diff
@@ -392,4 +399,37 @@ default String clueNotifyMessage() {
...
 
+    @ConfigItem(
+        keyName = "speedrunEnabled",
+        name = "Enable speedruns",
+        description = "Enable notifications for when you complete a speedrun",
+        position = 31,
+        section = speedrunSection
+    )
+    default boolean notifySpeedrun() {
+        return true;
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPluginConfig.java:410]: Should be false by default to be in line with all other notifiers

**Review by pajlada — CHANGES_REQUESTED** [review]
As discussed in chat, having the option to notify on every speedrun would be nice. Something like "Only notify for PBs", which would be enabled by default since the opposite is rather spammy

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 4]
```diff
@@ -271,5 +272,20 @@ public void onWidgetLoaded(WidgetLoaded event) {
...
+        final int SPEEDRUN_COMPLETED_PB_CHILD_ID = 12;
+        if (config.notifySpeedrun() && groupId == SPEEDRUN_COMPLETED_GROUP_ID) {
+            Widget questName = client.getWidget(SPEEDRUN_COMPLETED_GROUP_ID, SPEEDRUN_COMPLETED_QUEST_NAME_CHILD_ID);
+            Widget duration = client.getWidget(SPEEDRUN_COMPLETED_GROUP_ID, SPEEDRUN_COMPLETED_DURATION_CHILD_ID);
+            Widget personalBest = client.getWidget(SPEEDRUN_COMPLETED_GROUP_ID, SPEEDRUN_COMPLETED_PB_CHILD_ID);
+            if (questName == null || duration == null || personalBest == null) {
+                log.error("Found speedrun finished widget (group id {}) but it is missing something, questName={}, duration={}, pb={}", SPEEDRUN_COMPLETED_GROUP_ID, questName, duration, personalBest);
+            }
+//            log.info("quest name is {}, duration: {}, pb: {}", questName.getText(), duration.getText(), personalBest.getText());
+            this.speedrunNotifier.attemptNotify(Utils.parseQuestWidget(questName.getText()), duration.getText(), personalBest.getText());
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPlugin.java:288]:
```suggestion
            this.speedrunNotifier.attemptNotify(Utils.parseQuestWidget(questName.getText()), duration.getText(), personalBest.getText());
            this.levelNotifier.reset();
```

Comment (Mm2PL) [reply src/main/java/dinkplugin/DinkPlugin.java:288]: What does this do? Skip notifications?

Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPlugin.java:288]: Yes, if `LevelNotifier.currentLevels` is not cleared after going into a speedrun world, when the player hops to a normal world, the new levels will be received, which would look like a level up. Reset makes the skill levels null, so no notification occurs (`https://github.com/pajlads/DinkPlugin/blob/c48b91905eb96095fdf446376077b604db33413a/src/main/java/dinkplugin/LevelNotifier.java#L83`)

`levelNotifier.reset()` should also be called if the speedrun is cancelled early (i.e., started but not completed & world hop occurs)

Comment (Mm2PL) [reply src/main/java/dinkplugin/DinkPlugin.java:288]: Actually maybe skipping notifications should be a separate PR

**Review by pajlada — APPROVED** [review]

**Pull request merged**